logicrunner/executionbroker.go

```diff
@@ -209,6 +216,7 @@ func (q *ExecutionBroker) finishTask(ctx context.Context, transcript *Transcript
 		logger.Error("[ ExecutionBroker.FinishTask ] task '%s' is not in current", transcript.RequestRef.String())
 	} else {
 		q.currentList.Delete(transcript.RequestRef)
+		q.executionArchive.Done(transcript)
```

Comment:
q.currentList.Delete(transcript.RequestRef) and q.executionArchive.Done(transcript) aren't in transaction, can it cause some bugs?

Reply:
They aren't in transaction, but that's done under stateLock, so I don't see if it can cause any trouble.